core/src/display_object/avm1_button.rs

```diff
@@ -433,16 +433,27 @@ impl<'gc> TInteractiveObject<'gc> for Avm1Button<'gc> {
     fn filter_clip_event(
         self,
         _context: &mut UpdateContext<'_, 'gc>,
-        _event: ClipEvent,
+        event: ClipEvent,
     ) -> ClipEventResult {
         // An invisible button can still run its `rollOut` or `releaseOutside` event.
         // A disabled button doesn't run its events (`KeyPress` being the exception) but
         // its state can still change. This is tested at "avm1/mouse_events_visible_enabled".
         if !self.visible() && self.0.read().state == ButtonState::Up {
-            ClipEventResult::NotHandled
-        } else {
-            ClipEventResult::Handled
+            return ClipEventResult::NotHandled;
+        }
+
+        // The `keyPress` event doesn't fire if the button is inside another button.
+        if matches!(event, ClipEvent::KeyPress { .. })
+            && self
+                .base()
+                .parent
```

Comment:
(Nit: Can buttons be nested multiple levels inside each other? Such that one could test ancestors instead of just the parent? But I could imagine that such a case would be rather rare, and I think the changes look good as is).

Reply:
Yes they can, and it should work as expected with this PR. Here's [another sample](https://github.com/ruffle-rs/ruffle/files/11468171/testo.zip) with an addional level of button.